### ISSUE TITLE
Animations and animation state graphs do not load properly with templates

### ISSUE THREAD

**Issue by Astra307** [playcanvas/engine#4696]
After loading a template at run-time into the scene, animated objects in the template do not play. 

The issue seems to be that the animation state graph and animation assets that are linked to the objects in the template are not loaded together with the template or no loaded in the right order. Pre-loading these solves the problem, but is not ideal in terms of optimization and can be messy to keep track of.

**Comment by yaustar** [comment]
Do you have a repro project we can take a look at please?

Generally, the template is a separate asset to the assets that are referenced so when the template is instantiated, the components should attempt to lazy load the assets used in each component.

It sounds like this may not be happening correctly with the anim component 🤔

**Comment by Astra307** [comment]
Yeah, that's my impression. I'll try to reproduce it in a fresh project now.

**Comment by Astra307** [comment]
Hey, managed to reproduce it now in this project: https://playcanvas.com/editor/scene/1552117

**Comment by Astra307** [comment]
Hey, any progress on this issue?

**Comment by yaustar** [comment]
@ellthompson I've had another look at this and it looks like the anim assets are being loaded when the template is instanced into the scene but the anim system is not started/animating when it is done. 

Would the recent PR you've done for empty states also fix this as a side effect?

**Comment by Astra307** [comment]
hmm.. interesting. Actually it looks like calling anim.rebind() some time after  load makes the animations play! (but doing it when the assets load is marked "ready" seems to be too early)

**Comment by Astra307** [comment]
ok, my bad. That "hack" did not work. And this is still very much an issue. To get animations to play they have to be pre-loaded or manually loaded before a template using them are loaded. Otherwise they simply won't play

**Comment by yaustar** [comment]
> @ellthompson I've had another look at this and it looks like the anim assets are being loaded when the template is instanced into the scene but the anim system is not started/animating when it is done.
> 
> Would the recent PR you've done for empty states also fix this as a side effect? https://github.com/playcanvas/engine/pull/5112

I've just tried this with engine latest and it doesn't solve this particular issue.

@ellthompson Any ideas of workarounds in the short term?

**Comment by Astra307** [comment]
ok, I found a temporary hack that worked:
When loading a template that has animations attached to it, wait a bit, then re-load and instantiate the whole thing again.

**Comment by Astra307** [comment]
Hey, I noticed it looks like this issue has been fixed!

.. eexcept some issue with shape keys. 
If you immediately trigger an animation after the entity is instantiated it plays correctly, except for the shape keys. Briefly playing another animation later on or using the anim.rebind() function on a timeout fixes the problem, making the shape keys effect visible.
